Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -827,18 +827,34 @@ public final class ManifestLoader: ManifestLoaderProtocol {
                         // This provides some safety against arbitrary code execution when parsing manifest files.
                         // We only allow the permissions which are absolutely necessary.
                         if self.isManifestSandboxEnabled {
-                            let cacheDirectories = [self.databaseCacheDir, moduleCachePath].compactMap{ $0 }
-                            let strictness: Sandbox.Strictness = toolsVersion < .v5_3 ? .manifest_pre_53 : .default
-                            cmd = Sandbox.apply(command: cmd, strictness: strictness, writableDirectories: cacheDirectories)
+                            cmd = SandboxProfile(
+                                // Allow writing in the temporary directory (whether or not it's overridden by TMPDIR).
+                                .writable(localFileSystem.tempDirectory),
+
+                                // Allow writing in the database cache directory, if we have one.
+                                self.databaseCacheDir.map{ .writable($0) },
+
+                                // Allow writing in the module cache path, if there is one.
+                                moduleCachePath.map{ .writable($0) },
+
+                                // But do not allow writing in the directory that contains the manifest, even if it is
+                                // inside one of the writable directories.
+                                .readonly(manifestPath.parentDirectory)
```

Comment:
Do we have a better way of getting the package directory here?

Reply:
I don't think we do, but I think doing this is a fine assumption for now?

Reply:
It works OK for the purpose of sandboxing, but in general we could be parsing manifests that were just pulled from a Git repo, without having the whole package there.  Perhaps the package directory should be an optional directory path passed into the function.